packages/start/router.d.ts

```diff
@@ -0,0 +1,39 @@
+import { A as BaseA, Location, NavigateOptions, Navigator, RouteDataFunc } from "@solidjs/router";
+import { JSX } from "solid-js";
+import { LocationEntry, useSearchParams as useIslandsSearchParams } from "./islands/router";
+export type RouteParams<T extends string> = Record<T, string>;
+export type RouteDataArgs<T extends keyof StartRoutes = "$"> = {
+    data: StartRoutes[T]["data"];
+    params: RouteParams<StartRoutes[T]["params"]>;
+    location: Location;
+    navigate: Navigator;
+};
+declare const A: typeof BaseA;
+declare const Routes: (props: import("@solidjs/router").RoutesProps) => JSX.Element;
+declare const Outlet: () => JSX.Element;
+declare const useNavigate: () => Navigator;
+declare const useSearchParams: typeof useIslandsSearchParams | (<T extends import("@solidjs/router").Params>() => [T, (params: import("@solidjs/router").SetParams, options?: Partial<NavigateOptions<unknown>> | undefined) => void]);
+declare global {
+    interface Window {
+        router: {
+            navigate: (to: string, options?: Partial<NavigateOptions>) => Promise<boolean>;
+            push: (to: string | URL, options: Partial<NavigateOptions>) => void;
+            update: (body: string) => Promise<boolean>;
+            router: EventTarget;
+            location: () => LocationEntry;
+        };
+    }
+    interface StartRoutes {
+        $: {
+            params: any;
+            data: any;
+        };
+    }
+    interface Route {
+        "/notes/[note]": "/notes/[note]";
+    }
+}
```

Comment:
This file was generated from `Router.tsx`. It's used by the `tsconfig` via `"types": [ ..., "solid-start/router"`, which afaik only accepts `.d.ts` files ([or whole projects](https://www.typescriptlang.org/tsconfig#types)).

In actuality all `entry-client` needs is this here `declare global`, but I'm leaving it in `Router.tsx` for now for easy review. Could easily move it elsewhere if requested!